File: src/core/pipeline/search_drinks/drinks_client.py
import os

import aiohttp

DRINKS_API_URL = os.environ["DRINKS_API_URL"].rstrip("/")
DRINKS_API_KEY = os.environ.get("DRINKS_API_KEY", "")

CATEGORY_MAP = {
    "Cervezas": "beers",
    "Destilados": "spirits",
    "Vinos": "wines",
}

PACKAGING_MAP = {
    "Botella":   "Bottle",
    "Lata":      "Can",
    "Barril":    "Barrel",
    "Tetrapack": "Box/Tetrapack",
    "Caja":      "Box/Tetrapack",
    "Growler":   "Growler",
}


async def search(session: aiohttp.ClientSession, product: dict) -> list[dict]:
    """
    Query the drinks API by category, brand, abv, packaging and volume.
    Returns all matching drinks — caller is responsible for name filtering and selection.
    """
    category = CATEGORY_MAP.get(product.get("category", ""))
    if not category:
        print(f"[DrinksClient] Unknown category '{product.get('category')}' for '{product.get('name')}'")
        return []

    params = {
        "brand":     product["brand"],
        "minAbv":    product["abv"],
        "maxAbv":    product["abv"],
        "packaging": PACKAGING_MAP.get(product["packaging"], product["packaging"]),
        "minVolume": product["volume_ml"],
        "maxVolume": product["volume_ml"],
        "limit":     1000,
    }

    headers = {"X-Internal-Key": DRINKS_API_KEY} if DRINKS_API_KEY else {}

    async with session.get(f"{DRINKS_API_URL}/{category}", params=params, headers=headers) as resp:
        if resp.status == 200:
            return (await resp.json()).get("data", [])
        print(f"[DrinksClient] HTTP {resp.status} for '{product.get('name')}'")
        return []

File: src/pipeline/search-drinks/drinks-client.ts
const CATEGORY_MAP: Record<string, string> = {
    Cervezas: 'beers',
    Destilados: 'spirits',
    Vinos: 'wines',
}

const PACKAGING_MAP: Record<string, string> = {
    Botella: 'Bottle',
    Lata: 'Can',
    Barril: 'Barrel',
    Tetrapack: 'Tetrapack',
    Growler: 'Growler',
}

export type DrinkResult = {
    id: string
    name: string
    brand: string
    abv: number
    packaging: string
    volume: number
    [key: string]: unknown
}

export async function searchDrinks(product: {
    brand: string
    volumeMl: number
    abv: number
    packaging: string
    category: string
}): Promise<DrinkResult[]> {
    const apiCategory = CATEGORY_MAP[product.category]
    if (!apiCategory) return []

    const packaging = PACKAGING_MAP[product.packaging] ?? product.packaging
    const params = new URLSearchParams({
        brand: product.brand,
        minAbv: String(product.abv),
        maxAbv: String(product.abv),
        packaging,
        minVolume: String(product.volumeMl),
        maxVolume: String(product.volumeMl),
    })

    const baseUrl = process.env['DRINKS_API_URL']!.replace(/\/?$/, '/')
    const url = `${baseUrl}${apiCategory}?${params}`
    const response = await fetch(url, {
        headers: { 'X-Internal-Key': process.env['DRINKS_API_KEY']! },
    })

    if (!response.ok) return []
    return response.json() as Promise<DrinkResult[]>
}
